cxx-tests/src/tests.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use cxx::UniquePtr;
+use moveit::moveit;
+use moveit::Emplace;
+use moveit::EmplaceUnpinned;
+
+// Shared with C++
+const UNINITIALIZED: u8 = 0;
+const INITIALIZED: u8 = 1;
+const METHOD_CALLED: u8 = 2;
+
+#[cxx::bridge]
+mod ffi {
+  unsafe extern "C++" {
+    include!("cxx_support_test_cpp.h");
+    type Foo = super::bindgenish::Foo;
+    fn CreateUninitializedFoo() -> *mut Foo;
+    unsafe fn FreeUninitializedFoo(ptr: *mut Foo);
+
+    fn foo_constructor(_this: Pin<&mut Foo>);
+
+    fn get_status(self: &Foo) -> u8;
+    fn modify(self: Pin<&mut Foo>);
+  }
+  impl UniquePtr<Foo> {}
```

Comment:
Does this have any purpose other than ensuring a specific trait is implemented by Foo? If so, please include a comment.

Reply:
It does; done.